### PR TITLE
add servo, drive, touch, analog read

### PR DESCRIPTION
**Pull request by deanm1278**

### CODE REVIEW

File: drive.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            this._id = id;
+            this._pin = pin;
+        }
+
+        /**
+         * set the drive duty cycle
+         */
+        //% group="Drives"
+        //% weight=100
+        //% blockId=driveanalogwrite block="crickit set %pin value to %value=number"
```
Comment (pelikhan) [inline drive.ts]: drop the =number

Comment (pelikhan) [reply drive.ts]: we use "analog write ..." terminology in the CPX pins. we should align with that.
image

File: drive.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        //% group="Drives"
+        //% weight=100
+        //% blockId=drivesetfreq block="crickit set %pin frequency to %value=number"
+        //% value.min=0 value.max=2000
+        //% pin.fieldEditor="gridpicker"
+        //% pin.fieldOptions.width=220
+        //% pin.fieldOptions.columns=2
+        setFrequency(value: number) {
+            const dev = saw();
+            dev.setPwmFreq(this._pin, value);
```
Comment (pelikhan) [inline drive.ts]: check range

File: drive.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            value = Math.map(value, 0, 1023, CRICKIT_PWM_MIN, CRICKIT_PWM_MAX);
+            value = Math.clamp(CRICKIT_PWM_MIN, CRICKIT_PWM_MAX, value);
+            dev.analogWrite(this._pin, value);
+        }
+
+        /**
+         * set the drive frequency
+         */
+        //% group="Drives"
+        //% weight=100
```
Comment (pelikhan) [inline drive.ts]: weight=10 (needs to be lower than analogWrite)

File: drive.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        //% blockId=drivesetfreq block="crickit set %pin frequency to %value=number"
```
Comment (pelikhan) [inline drive.ts]: drop =number

File: servo.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            this._id = id;
+            this._pin = pin;
+        }
+
+        /**
+         * set the servo angle
+         */
+        //% group="Servos"
+        //% weight=100
+        //% blockId=sawservosetangle block="crickit set %pin angle to %value=number"
```
Comment (pelikhan) [inline servo.ts]: drop =number

Comment (pelikhan) [reply servo.ts]: %pin --> %servo

File: servo.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        //% value.min=0 value.max=180
+        //% pin.fieldEditor="gridpicker"
```
Comment (pelikhan) [inline servo.ts]: pin --> servo

**Review by pelikhan — APPROVED** [review]
Add touch.ts to pxt.json

**Comment by deanm1278** [comment]
ok @pelikhan changes have been made!

**Review by pelikhan — APPROVED** [review]

**Pull request merged**